internal/featuretests/v3/envoy.go

```diff
@@ -110,7 +110,7 @@ func routeCluster(cluster string, opts ...func(*envoy_route_v3.Route_Route)) *en
 	return r
 }
 
-func routePrefix(prefix string, headers ...dag.HeaderMatchCondition) *envoy_route_v3.RouteMatch {
+func routePrefixWithHeaderConditions(prefix string, headers ...dag.HeaderMatchCondition) *envoy_route_v3.RouteMatch {
```

Comment:
It looks like most uses of this function don't actually use any header conditions, apart from in `headercondition_test.go` -- for readability I'm thinking it'd be nice to keep a func named `routePrefix` that _only_ takes a prefix, and use that where appropriate, then have`routePrefixWithHeaderConditions` and `routePrefixWithQueryParameterConditions` separately to be used where needed.